gen-dockerfiles

```diff
@@ -7,27 +7,28 @@ RUST_VERSION     =  '1.71.1'
 RUSTUP_VERSION   =  '1.26.0'
 OPENSSL_VERSION  =  '3.1.3'
 
-RUST_ARCH = 'x86_64-unknown-linux-gnu'
-
-DISTRIBUTIONS = [
+DISTRIBUTIONS = {
     # Ubuntu
-    'bionic',
+    'bionic': 'x86_64-unknown-linux-gnu',
     # CentOS
-    'centos-7',
-]
+    'centos-7': 'x86_64-unknown-linux-gnu',
+    # Arch64
+    'aarch64': 'aarch64-unknown-linux-gnu',
+}
 
 
-def fetch_rustup_hash():
+def fetch_rustup_hash(RUST_ARCH):
     url = f'https://static.rust-lang.org/rustup/archive/{RUSTUP_VERSION}/{RUST_ARCH}/rustup-init.sha256'
     with request.urlopen(url) as f:
-        return f.read().decode('utf-8').split()[0]
+        res = f.read().decode('utf-8').split()[0]
+        print("res: {}", res)
+        return res
 
 def fetch_openssl_hash():
-    url = f'https://www.openssl.org/source/openssl-{OPENSSL_VERSION}.tar.gz.sha256'
+    url = f'http://www.openssl.org/source/openssl-{OPENSSL_VERSION}.tar.gz.sha256'
```

Comment:
It's unsafe to fetch sha256 via http

Reply:
oh, forget this temporary change. fixed.